src/rt/array/capacity.d

```diff
@@ -195,3 +195,98 @@ auto ref inout(T[]) assumeSafeAppend(T)(auto ref inout(T[]) arr) nothrow @system
     assert(is(typeof(a3) == int[]));
     assert(is(typeof(b3) == immutable(int[])));
 }
+
+// HACK: `nothrow` and `pure` is faked.
+private extern (C) void[] _d_arraysetlengthT(const TypeInfo ti, size_t newlength, void[]* p) nothrow pure;
+private extern (C) void[] _d_arraysetlengthiT(const TypeInfo ti, size_t newlength, void[]* p) nothrow pure;
+
+/**
+ * Resize dynamic array
+ * Params:
+ *  a = the array that will be resized, taken as a reference
+ *  newlength = new length of array
+ * Returns:
+ *  The new value of `arr`
+ * Bugs:
+ *   The safety level of this function is faked. It shows itself as `@trusted pure nothrow` to not break existing code.
+ */
+Tarr _d_arraysetlengthT(Tarr : T[], T)(return scope ref Tarr arr, size_t newlength) @trusted
+{
+    version (D_BetterC)
```

Comment:
I'm trying to figure out why this check is necessary, but I haven't succeeded yet.  It shouldn't be required, but is a successful workaround.  Until then, please use `D_TypeInfo` instead of `D_BetterC`.  I think GDC does not support betterC, but instead has individual flags like `-fno-rtti`, `-fno-exceptions`, etc.  Using the more specific version specifier should make this code more portable.  The error message would also need to be updated e.g. "Cannot resize arrays if compiling without support for runtime type information."

Reply:
Thanks, I will do the same changes on the other PRs.